crates/ethereum/evm/src/execute.rs

```diff
@@ -698,7 +694,8 @@ mod tests {
 
         let provider = executor_provider(chain_spec);
 
-        let mut executor = provider.batch_executor(StateProviderDatabase::new(&db));
+        let mut executor =
+            provider.batch_executor(StateProviderDatabase::new(&db), PruneModes::none());
```

Comment:
should `fn batch_executor` take `T: Into<StateProviderDatabase>`? so that we don't need to do this wrapping everywhere? and perhaps blanket implement `From<DB: Database> for StateProviderDatabase<DB>`?

Reply:
no this needs to operate on DB: Database